aiomcache/client.py

```diff
@@ -58,6 +59,70 @@ def _validate_key(self, key):
         return key
 
     @asyncio.coroutine
+    def close(self):
+        """Closes the sockets if its open."""
+        yield from self._pool.clear()
+
+    @acquire
+    def get(self, conn, key, default=None, with_flags=False):
+        """Gets a single value from the server.
+
+        :param key: ``bytes``, is the key for the item being fetched
+        :param default: default value if there is no value.
+        :return: ``bytes``, is the data for this specified key.
+        """
+        values, _, flags = yield from self._multi_get(
+            conn, key, with_cas=False, with_flags=with_flags)
+        if with_flags:
+            return values.get(key, default), flags.get(key)
+        else:
+            return values.get(key, default)
+
+    @acquire
+    def gets(self, conn, key, default=None, with_flags=False):
+        """Gets a single value from the server together with the cas token.
+
+        :param key: ``bytes``, is the key for the item being fetched
+        :param default: default value if there is no value.
+        :return: ``bytes``, ``bytes tuple with the value and the cas
+        """
+        values, cas_tokens, flags = yield from self._multi_get(
+            conn, key, with_cas=True, with_flags=with_flags)
+
+        if with_flags:
+            return values.get(key, default), cas_tokens.get(key), \
+                flags.get(key)
+        else:
+            return values.get(key, default), cas_tokens.get(key)
+
+    @acquire
+    def multi_get(self, conn, *keys, with_cas=False, with_flags=False):
+        """Takes a list of keys and returns a list of values.
+
+        :param keys: ``list`` keys for the item being fetched.
+        :return: ``list`` of values for the specified keys.
+        :raises:``ValidationException``, ``ClientException``,
+        and socket errors
+        """
+        values, cas_tokens, flags = yield from self._multi_get(
+            conn, *keys, with_cas=with_cas, with_flags=with_flags)
+        if with_cas:
```

Comment:
looks like user can specify `with_cas` and `with_flags` in same time, should we raise exception in this case? otherwise result may be unexpected

Reply:
Is there any case when it is useful to return value, flags and cas?

Reply:
Yes - for example, the flags may be used to specify how the value is encoded, so the caller may want both the (decoded) value and cas token. Passing with_cas=True and with_flags=True should work as expected; this is tested in test_set_flags.

Reply:
oh, I misread code sorry, make sense